src/GovUk.Education.ExploreEducationStatistics.Admin/Services/DataBlockService.cs

```diff
@@ -122,9 +122,32 @@ public class DataBlockService : IDataBlockService
                 {
                     var viewModel = _mapper.Map<DataBlockViewModel>(dataBlock);
 
+                    var subjectId = dataBlock.Query.SubjectId;
+
+                    var releaseContentBlock = await _context.ReleaseContentBlocks
+                        .Include(rcb => rcb.Release)
+                        .SingleOrDefaultAsync(rcb => rcb.ContentBlockId == id);
+
+                    var releaseId = releaseContentBlock.ReleaseId;
+
+                    viewModel.DataSetId = await _context.ReleaseFiles.Include(rf => rf.File)
+                        .Where(rf =>
+                            rf.ReleaseId == releaseId &&
+                            rf.File.SubjectId == subjectId &&
+                            rf.File.Type == FileType.Data)
+                        .Select(rf => rf.File.Id).SingleAsync();
+
+                    viewModel.DataSetName = await _context.ReleaseFiles.Include(rf => rf.File)
+                        .Where(rf =>
+                            rf.ReleaseId == releaseId &&
+                            rf.File.SubjectId == subjectId &&
+                            rf.File.Type == FileType.Data)
+                        .Select(rf => rf.Name).SingleAsync();
```

Comment:
Even though data files will all have a name, `rf.Name` is optional (because `ReleaseFile` is used for other types of files which don't all have a name), so you need to add a fall back to something like the empty string.